lib/graphql/object_type.rb

```diff
@@ -42,4 +42,26 @@ def interfaces=(new_interfaces)
   def kind
     GraphQL::TypeKinds::OBJECT
   end
+
+  # @return [GraphQL::Field] The field definition for `field_name` (may be inherited from interfaces)
+  def get_field(field_name)
+    fields[field_name] || get_interface_field(field_name)
+  end
+
+  # @return [Array<GraphQL::Field>] All fields, including ones inherited from interfaces
+  def all_fields
+    interface_fields = interfaces.inject({}) do |memo, iface|
+      memo.merge(iface.fields)
+    end
+    interface_fields.merge(self.fields).values
+  end
+
+  private
+
+  # Find the _last_ definition for `field_name`
+  def get_interface_field(field_name)
+    interfaces.reduce(nil) do |memo, iface|
```

Comment:
Oh, now I understand, yes, that's much simpler! 

I think I'll still use the block for merging though. I could add `InterfaceType#merge` for this purpose but I don't think it's worth expanding the API